interp/constrexpr_ops.ml

```diff
@@ -390,8 +389,8 @@ let fold_constr_expr_with_binders g f n acc = CAst.with_val (function
       let acc = Option.fold_left (f (Id.Set.fold g ids n)) acc rtnpo in
       let acc = List.fold_left (f n) acc (List.map (fun (fst,_,_) -> fst) al) in
       List.fold_right (fun {CAst.v=(patl,rhs)} acc ->
-          let ids = ids_of_pattern_list patl in
-          f (Id.Set.fold g ids n) acc rhs) bl acc
+          let (n,acc) = List.fold_left (List.fold_left (cases_pattern_fold_names g (fun (n,acc) t -> (n,f n acc t)))) (n,acc) patl in
+          f n acc rhs) bl acc
```

Comment:
The new code sounds much less readable than the previous one?
Also not sure how this (and the few previous edits in this commit) is related to the commit message "Adding computation of free variables in CPatCast."

Reply:
Ok, but the new code is really hard to read. Maybe breaking it in successive let in giving proper names to arguments could help. For instance there are f and g everywhere, what are they, sure functions, but expected to do what?

Reply:
I pushed an extra commit of `constrexpr_ops.ml` that document and uniformize the notations around `g`, `f`, `acc`, `e`, `n`, etc. (not rebased on #19561 yet though). Tell me if you think that more should be done.